### PR TITLE
`services/regulated-assets-approval-server/internal/serve/`:  Add validator and tests to toml handler

### PR DESCRIPTION
**Pull request by Zagan202**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [X] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [X] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [X] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [X] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Adds new tests to the TOML handler and adds a validate function to ensure correct data is given.
Closes: https://github.com/stellar/nebula/issues/184
### Why

Improve test coverage.

### Known limitations

N/A

### CODE REVIEW

File: services/regulated-assets-approval-server/internal/serve/toml_test.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	}
+	err = h.validate()
+	require.EqualError(t, err, "approval server cannot be empty")
+
+	// empty kyc threshold
+	h = stellarTOMLHandler{
+		networkPassphrase: network.TestNetworkPassphrase,
+		assetCode:         "FOOBAR",
+		issuerAddress:     "GCVDOU4YHHXGM3QYVSDHPQIFMZKXTFSIYO4HJOJZOTR7GURVQO6IQ5HM",
+		approvalServer:    "localhost:8000/tx_approve",
```
Comment (marcelosalloum) [inline services/regulated-assets-approval-server/internal/serve/toml_test.go:60]: 💡  Could you plz replace all occurrences of `tx_approve` with `tx-approve` for consistency? I believe we've decided to go with the latter, right?

File: services/regulated-assets-approval-server/internal/serve/toml.go [hunk 2]
```diff
@@ -13,8 +18,45 @@ type stellarTOMLHandler struct {
...
+		return errors.New("asset issuer address is not a valid public key")
+	}
+
+	if h.approvalServer == "" {
+		return errors.New("approval server cannot be empty")
+	}
+
+	if h.kycThreshold == 0 {
+		return errors.New("kyc threshold cannot be zero")
+	}
```
Comment (marcelosalloum) [inline services/regulated-assets-approval-server/internal/serve/toml.go:44]: 💡  Taking another look at this validation I can see negative thresholds would be mistakenly accepted, so we should update this validation to make sure only positive values pass.

**Review by marcelosalloum — APPROVED** [review]
LGTM! I've added 2 improvement suggestions below ⬇️

File: services/regulated-assets-approval-server/internal/serve/toml_test.go (hunk 1 above)
Comment (Zagan202) [reply services/regulated-assets-approval-server/internal/serve/toml_test.go]: Addressed here ec14213 for `toml_test.go`

I'll take care of the other occurrences in the `tx_approve.go` port PR

File: services/regulated-assets-approval-server/internal/serve/toml.go [hunk 3]
```diff
@@ -13,8 +18,45 @@ type stellarTOMLHandler struct {
...
+		return errors.New("asset issuer address is not a valid public key")
+	}
+
+	if h.approvalServer == "" {
+		return errors.New("approval server cannot be empty")
+	}
+
+	if h.kycThreshold <= 0 {
+		return errors.New("kyc threshold cannot be less than or equal to zero")
+	}
```
Comment (marcelosalloum) [inline services/regulated-assets-approval-server/internal/serve/toml.go:44]: Could you additionally update the tests to make sure the app is protected against negative values?

Comment (Zagan202) [reply services/regulated-assets-approval-server/internal/serve/toml.go:44]: We should be good to go 0e53a95

**Pull request merged**